src/guides/v2.3/architecture/tech-stack.md

```diff
@@ -63,8 +63,11 @@ Magento 2.2+ does not support HipHop Virtual Machine (HHVM).
 
 Magento also provides automated testing suites that include unit, integration, functional and performance test scripts, as well as JavaScript tests and tools for static code analysis. Components include PHPUnit for the unit test framework and Selenium for the functional test framework.
 
-This framework is located in the `dev/tests` directory. The functional testing framework `mtf` can be found in a [separate repository](https://github.com/magento/mtf){:target="_blank"}.
-For more information, see the [Functional Testing Framework]({{page.baseurl}}/mtf/mtf_introduction.html) guide.
+This framework is located in the `<magento_root>/app/code/<vendor_name>/<module_name>/Test/Mftf/` for local, customized tests and `<magento_root>/vendor/<vendor_name>/<module_name>/Test/Mftf/` for tests provided by Magento and vendors. If you installed Magento with Composer, please refer to vendor/magento/<module_dir>/Test/Mftf/ for examples.
```

Comment:
This is a set of frameworks, as I understand. Each directory in [dev/tests](https://github.com/magento/magento2/tree/2.4-develop/dev/tests) contains a framework of a particular type of testing.

Regarding functional framework, since `functional/` directory was occupied with old FTF, the MFTF was put to a new `acceptance/` directory. Functional tests were stored there at first, but later they were moved closer to code, to actual Magento modules. The MFTF as a framework itself was moved to be packaged as a composer dependency. But the `acceptance/` directory still contains some files used by the framework for integration with Magento (I guess). Many things could change, so @dobooth may tell more details on this, I think.

As to actual tests that are run by corresponding frameworks, some types of tests are located near modules at _`<magento_root>/app/code/<vendor_name>/<module_name>/Test/_ (for example, MFTF and Unit tests), and some of them are still located near framework such as integration tests in `magento2/dev/tests/integration/testsuite/Magento/<module_name>`.

[Magento Testing Guide](https://devdocs.magento.com/guides/v2.3/test/testing.html) contains some overview about all this. So, the text may refer to it.

This paragraph can be changed to something like:

_Testing frameworks are located in the `dev/tests` directory. Refer to our [Magento Testing Guide](https://devdocs.magento.com/guides/v2.3/test/testing.html) to learn more about the frameworks and associated tests._

Reply:
Accepted suggestion